app/labor/markdown_parser.rb

```diff
@@ -18,8 +19,9 @@ def finalize(link_attributes: {})
     html = markdown.render(escaped_content)
     sanitized_content = sanitize_rendered_markdown(html)
     begin
-      parsed_liquid = Liquid::Template.parse(sanitized_content)
-      html = markdown.render(parsed_liquid.render(nil, registers: { source: @source }))
+      liquid_tag_options = { source: @source, user: @user }
+      parsed_liquid = Liquid::Template.parse(sanitized_content, liquid_tag_options)
```

Comment:
Digging through the [`liquid` gem](https://github.com/Shopify/liquid) source code, I found that [the second argument here is actually `options`](https://github.com/Shopify/liquid/blob/1feaa6381300d56e2c71b49ad8fee0d4b625147b/lib/liquid/template.rb#L133). When you continue to pull on that thread, you'll find these options become a `ParsedContext` object which is then [passed to the `initialize` method by default](https://github.com/thepracticaldev/dev.to/pull/8917/files#r445812102).

Reply:
so the `registers:` options is not needed anymore?

Reply:
Correct! We are passing the same information in an improved way (in the initialize step and using options).